packages/xo-server/src/xapi/index.mjs

```diff
@@ -663,21 +663,29 @@ export default class Xapi extends XapiBase {
 
     const token = await hostXapi.call('host.migrate_receive', host.$ref, migrationNetwork.$ref, {})
 
-    const loop = () =>
-      this.callAsync(
-        'VM.migrate_send',
+    const loop = () => {
+      const params = [
         vm.$ref,
         token,
         true, // Live migration.
         vdis,
         vifsMap,
         {
           force: force ? 'true' : 'false',
-        }
+        },
         // FIXME: missing param `vgu_map`, it does not cause issues ATM but it
         // might need to be changed one day.
         // {},
-      )::pCatch({ code: 'TOO_MANY_STORAGE_MIGRATES' }, () => pDelay(1e4).then(loop))
+      ]
+
+      return this.callAsync('VM.assert_can_migrate', ...params)
```

Comment:
IMHO, it does not make sense to do it in the loop, we should probably do it only once before.